cmd/soroban-rpc/internal/jsonrpc.go

```diff
@@ -53,16 +60,49 @@ func NewJSONRPCHandler(cfg *config.LocalConfig, params HandlerParams) Handler {
 		"getTransaction":      methods.NewGetTransactionHandler(params.TransactionStore),
 		"sendTransaction":     methods.NewSendTransactionHandler(params.Logger, params.TransactionStore, cfg.NetworkPassphrase, params.CoreClient),
 		"simulateTransaction": methods.NewSimulateTransactionHandler(params.Logger, params.LedgerEntryReader, params.PreflightGetter),
-	}, nil)
-	corsMiddleware := cors.New(cors.Options{
-		AllowedOrigins: []string{"*"},
-		AllowedHeaders: []string{"*"},
-		AllowedMethods: []string{"GET", "PUT", "POST", "PATCH", "DELETE", "HEAD", "OPTIONS"},
-	})
+	}, &bridgeOptions)
 
 	return Handler{
 		bridge:  bridge,
 		logger:  params.Logger,
-		Handler: corsMiddleware.Handler(bridge),
+		Handler: bridge,
 	}
 }
+
+type rpcLogger struct {
+	logger *log.Entry
+}
+
+func (r *rpcLogger) LogRequest(ctx context.Context, req *jrpc2.Request) {
+	logger := r.logger.WithFields(log.F{
+		"subsys": "jsonrpc",
+		// FIXME: the HTTP request context is independent from the JSONRPC context, and thuse the code below doesn't work
+		// "req":      middleware.GetReqID(ctx),
```

Comment:
Thought: in opentracing terminology, it seems like the jsonrpc handler would be a subspan of the http handler. 🤔

Reply:
Exactly what I thought. The problem is we don’t have a way to join the spans